File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int curSum = 0, count = 0;
        unordered_map<int, int> mp;
        mp[0] = 1;

        for (auto n : nums) {
            curSum += n;
            if (mp.contains(curSum - k))
                count += mp[curSum - k];
            mp[curSum]++;
        }
        return count;
    }
};

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.go
func subarraySum(nums []int, k int) int {
    prefix := 0
    count := 0
    mp := map[int]int {prefix:1}

    for _, v := range nums {
        prefix += v
        find := prefix - k
        count += mp[find]
        mp[prefix]++
    }

    return count
}

